src/vbe_info.rs

```diff
@@ -115,50 +216,103 @@ impl fmt::Debug for VBEModeInfo {
+
+        /// Is the controller VGA compatible.
         const NOT_VGA_COMPATIBLE = 0x2; // The controller can be switched into VGA modes.
+
+        /// The operating behaviour of the RAMDAC
         const RAMDAC_FIX = 0x4;         // When writing lots of information to the palette, the blank bit should be used.
     }
 }
 
 bitflags! {
+    /// A Mode attributes bitfield.
     pub struct VBEModeAttributes: u16 {
+        /// Mode supported by hardware configuration.
         const SUPPORTED = 0x1;           // This mode is supported by the hardware.
+
+        /// TTY Output functions supported by BIOS
         const TTY_SUPPORTED = 0x4;       // TTY output is supported.
+
+        /// Color support.
         const COLOR = 0x8;
+
+        /// Mode type (text or graphics).
         const GRAPHICS = 0x10;
+
+        /// VGA compatibility.
         const NOT_VGA_COMPATIBLE = 0x20;
-        const NO_VGA_WINDOW = 0x40;      // If this is set, the window A and B fields of VBEModeInfo are invalid.
-        const LINEAR_FRAMEBUFFER = 0x80; // A linear framebuffer is available for this mode.
+
+        /// VGA Window compatibility.
+        ///
+        /// If this is set, the window A and B fields of VBEModeInfo are invalid. 
+        const NO_VGA_WINDOW = 0x40;
+
+        /// Linear framebugger availability.
+        ///
+        /// Set if a linear framebuffer is available for this mode.
+        const LINEAR_FRAMEBUFFER = 0x80;
     }
 }
 
 bitflags! {
+    /// The WindowAttributes describe the characteristics of the CPU windowing
+    /// scheme such as whether the windows exist and are read/writeable, as follows:
     pub struct VBEWindowAttributes: u8 {
+        /// Relocatable window(s) supported?
         const RELOCATABLE = 0x1;
+
+        /// Window is readable?
         const READABLE = 0x2;
+
+        /// Window is writeable?
         const WRITEABLE = 0x4;
     }
 }
 
 bitflags! {
+
+    /// The DirectColorModeInfo field describes important characteristics of direct color modes.
+    ///
+    /// Bit D0 specifies whether the color ramp of the DAC is fixed or
+    /// programmable. If the color ramp is fixed, then it can not be changed.
+    /// If the color ramp is programmable, it is assumed that the red, green,
+    /// and blue lookup tables can be loaded by using VBE Function 09h
+    /// (it is assumed all color ramp data is 8 bits per primary).
+    /// Bit D1 specifies whether the bits in the Rsvd field of the direct color
+    /// pixel can be used by the application or are reserved, and thus unusable.
     pub struct VBEDirectColorAttributes: u8 {
-        const PROGRAMMABLE = 0x1;       // The color ramp of the DAC is programmable.
-        const RESERVED_USABLE = 0x2;    // The bits of the 'reserved' field are usable by the application.
+        /// Color ramp is fixed/programmable
+        const PROGRAMMABLE = 0x1;
+
+        /// Bits in Rsvd field are usable/reserved
```

Comment:
Can you clarify that when the flag is cleared/zero, the bits in Rsvd field are reserved? And when the flag is set/one, the bits in Rsvd field are usable?